numba/typing/templates.py

```diff
@@ -294,6 +289,101 @@ class _OverloadFunctionTemplate(AbstractTemplate):
     A base class of templates for overload functions.
     """
 
+    def _validate_sigs(self, typing_func, impl_func):
```

Comment:
I think this method shouldn't be raising `TypingError`.  The error is not a typing issue but a mistake in the implementation.  We should raise something like `numba.errors.InternalError`.

Reply:
Done in 32ef66f, note however that because this error is raised during typing it will end up being wrapped in a `TypingError` class.

Reply:
We can fine tune the exception catching/rewrite later when we differentiate between InternalError and TypingError (or UserError) more.  InternalError should cause a panic in the compiler and stop it as quickly as possible.